--- gcc/gcc/ggc-common.c
+++ gcc/gcc/ggc-common.c
@@ -29,7 +29,9 @@
 #include "params.h"
 #include "hosthooks.h"
 #include "plugin.h"
+#ifdef HAVE_SYS_RESOURCE_H
 #include <sys/resource.h>
+#endif
 
 /* When set, ggc_collect will do collection.  */
 bool ggc_force_collect;
@@ -729,7 +729,6 @@ static double
 ggc_rlimit_bound (double limit)
 {
-#if defined(HAVE_GETRLIMIT)
-// && !defined __amiga__
+#if defined(HAVE_GETRLIMIT) && !defined(__amigaos__)
   struct rlimit rlim;
 # if defined (RLIMIT_AS)
   /* RLIMIT_AS is what POSIX says is the limit on mmap.  Presumably
@@ -796,7 +795,7 @@ ggc_min_heapsize_heuristic (void)
      bound of 128M (when RAM >= 1GB).  */
   phys_kbytes /= 8;
 
-#if defined(HAVE_GETRLIMIT) && defined (RLIMIT_RSS)
+#if defined(HAVE_GETRLIMIT) && defined (RLIMIT_RSS) && !defined(__amigaos__)
   /* Try not to overrun the RSS limit while doing garbage collection.
      The RSS limit is only advisory, so no margin is subtracted.  */
  {
