--- libbfd-in.h
+++ libbfd-in.h
@@ -194,7 +194,7 @@ extern bool _bfd_bool_bfd_link_true
 extern bool _bfd_bool_bfd_bfd_true
   (bfd *, bfd *) ATTRIBUTE_HIDDEN;
 extern bool _bfd_bool_bfd_uint_true
-  (bfd *, unsigned int) ATTRIBUTE_HIDDEN;
+  (bfd *, flagword) ATTRIBUTE_HIDDEN;
 extern bool _bfd_bool_bfd_asection_bfd_asection_true
   (bfd *, asection *, bfd *, asection *) ATTRIBUTE_HIDDEN;
 extern bool _bfd_bool_bfd_asymbol_bfd_asymbol_true
