--- a/include/sys/statvfs.h	2026-08-17 17:44:12.458215539 +0900
+++ b/include/sys/statvfs.h	2026-08-17 17:44:16.259161554 +0900
@@ -35,6 +35,8 @@
 #define ST_NOSUID         0x0002 /* Does not support the semantics of the ST_ISUID and ST_ISGID file mode bits. */
 #define ST_CASE_SENSITIVE 0x0004 /* The file system is case sensitive. */
 
+int statvfs(const char *path, struct statvfs *buf);
+
 #ifdef __cplusplus
 }
 #endif
